src/Engine/Objects/SpriteObject.h

```diff
@@ -14,6 +14,7 @@ struct SpriteObject {
         return uAttributes & SPRITE_ATTACHED_TO_HEAD;
     }
 
+    // WARNING - this function should not be called from any item in pSpriteObjects vector
```

Comment:
`from` -> `for`?

Why don't we add an assert inside?

Or, why don't we make it into a static function?

Both can be left as TODOs for now, but let's spell this out.

Reply:
I thought assert was a bit too brute force and static function is a no go because we use 'this' within.

Will add in the assert option for now and leave a Todo for a refactor later.